### PR TITLE
More block: use an actual placeholder for input text.

### PR DESCRIPTION
**Pull request by ZebulanStanphill**
## Description
The More block has been using this weird default text system for a while. This PR updates it to just use a standard placeholder, eliminating the weirdness around when the theme's default text is used on the front-end or not. It also adds an `aria-label` to the label input, improving accessibility.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by draganescu** [comment]
Howdy @ZebulanStanphill, what problem does this fix?

**Comment by ZebulanStanphill** [comment]
@draganescu As discovered during the work in #23758, the More block was showing some default text that didn't always match the theme default (as is the case with Twenty Twenty), and if you modified this text at all, you wouldn't be able to get back to the default without erasing the whole thing, and this would display as empty until refresh, resulting in inconsistent behavior. The whole thing was fragile and inconsistent with other blocks that show placeholders, such as the Heading or Button blocks.

Basically, the More block was showing a placeholder by actually filling it with content upon insertion, rather than a proper placeholder. That's not accessible and it's inconsistent with other blocks. The logic for determining whether or not to display the theme default on the front-end or not was really fragile as a result. This PR fixes all that.

**Comment by afercia** [comment]
> what problem does this fix?

From an accessibility perspective, technically, the block uses an unlabelled input field, which is bad:

`<input type="text" value="Read more" style="width: 10.2em;">`

@ZebulanStanphill regarding the aria label value: not sure the text `Read more link label` is 100% appropriate. Maybe `Read more link text` would be more understandable for users (and also technically more correct)?

Aside:
I'm not sure why the block name is only "More". This isn't clear and I think it's confusing for users as this has been universally known for years as "Read more" link. Also, there are too many things named "More" in Gutenberg thus using the classic "Read more" name would reduce potential confusion.

Current block name:

Screenshot 2020-09-19 at 17 00 39


Read more link in Classic Editor (where the term "tag" is a bit incorrect):

Screenshot 2020-09-19 at 17 02 37

**Review by afercia — APPROVED** [review]
From an accessibility perspective, this fixes a bug where an input field isn't labelled at all. Totally in favor of the a11y improvements here. Will leave code review to others.

**Review by draganescu — APPROVED** [review]
@ZebulanStanphill great I see now. I tested this and works as described. With your explanation I think the code changes make sense. 🚢 it!
